src/ap/java/org/spongepowered/processor/event/ListenerProcessor.java

```diff
@@ -0,0 +1,105 @@
+
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic;
+
+@SupportedAnnotationTypes(ListenerProcessor.LISTENER_ANNOTATION_CLASS)
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+public class ListenerProcessor extends AbstractProcessor {
+
+    static final String LISTENER_ANNOTATION_CLASS = "org.spongepowered.api.event.Listener";
+    private static final String EVENT_CLASS = Event.class.getName();
+
+    @Override
+    public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        if (ProcessorUtils.contains(annotations, Listener.class)) {
+            for (Element e : roundEnv.getElementsAnnotatedWith(Listener.class)) {
+                if (e.getKind() != ElementKind.METHOD) {
+                    processingEnv.getMessager().printMessage(ERROR, "Invalid element of type " + e.getKind() + " annotated with @Listener", e);
+                    continue;
+                }
+                ExecutableElement method = (ExecutableElement) e;
+
+                Messager msg = this.processingEnv.getMessager();
+                if (method.getModifiers().contains(Modifier.STATIC)) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must not be static", method);
+                }
+                if (!method.getModifiers().contains(Modifier.PUBLIC)) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must be public", method);
+                }
+                if (method.getModifiers().contains(Modifier.ABSTRACT)) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must not be abstract", method);
+                }
+                if (method.getEnclosingElement().getKind().isInterface()) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "interfaces cannot declare listeners", method);
+                }
+                if (method.getReturnType().getKind() != TypeKind.VOID) {
+                    msg.printMessage(Diagnostic.Kind.ERROR, "method must return void", method);
+                }
+                List<? extends VariableElement> parameters = method.getParameters();
+                if (parameters.isEmpty() || isTypeSubclass(parameters.get(0), EVENT_CLASS)) {
```

Comment:
This should be `|| !isTypeSubClass(...)` because you want to show the error if the method doesn't have an `Event` as first parameter